common/internal_model/src/main/scala/weco/catalogue/internal_model/work/Relations.scala

```diff
@@ -59,18 +60,11 @@ case class Relations(
   * Series ancestor is to be discarded.
   */
 object RelationSet {
-  def apply(relations: List[Relation]): List[Relation] = {
-    // Store the original first positions, in order to preserve order.
-    // Order is significant in RelationSets, e.g. in a hierarchy, ancestors = parent,grandparent,great grandparent etc.
-    // For hierarches, this could be determined by depth, but in the case of Series links,
-    // the order is determined by the order in the source data.
-    val indexMap = relations.map(_.title).distinct.zipWithIndex.toMap
-    relations
-      .groupBy(_.title)
-      .values
-      .map(_.maxBy(_.id))
-      .toList
-      .sortBy(relation => indexMap(relation.title))
+  def apply(existingRelations: List[Relation],
+            newRelations: List[Relation]): List[Relation] = {
+    val newTitles = newRelations.map(_.title).distinct
```

Comment:
nit - not sure you need the `distinct`? Would that we had `distinctBy` in 2.12 for this whole block